File: examples/tool_permission_workflow_test.py
from tool_permission_plan import plan_repair_followup


def test_commits_repair_when_agent_edited_code_and_tests_passed() -> None:
    output = {
        "summary": "Read src/app.py, found 1 related matches, and ran pytest.",
        "searched": True,
        "test_passed": True,
        "edited": True,
    }

    assert plan_repair_followup(output) == {
        "action": "commit_repair",
        "requires_human_review": False,
        "summary": output["summary"],
    }


def test_reruns_tests_when_edits_happened_but_verification_failed() -> None:
    output = {
        "summary": "Edited src/app.py but the focused test failed.",
        "searched": True,
        "test_passed": False,
        "edited": True,
    }

    assert plan_repair_followup(output) == {
        "action": "rerun_tests",
        "requires_human_review": True,
        "summary": output["summary"],
    }


if __name__ == "__main__":
    test_commits_repair_when_agent_edited_code_and_tests_passed()
    test_reruns_tests_when_edits_happened_but_verification_failed()

File: examples/tool_permission_workflow.test.ts
import { describe, expect, it } from "vitest";
import { planRepairFollowup, type ToolRepairOutput } from "./tool_permission_plan.js";

describe("tool permission workflow typed output planning", () => {
  it("commits a repair when the tool-backed agent edited code and tests passed", () => {
    const output: ToolRepairOutput = {
      summary: "Read src/app.ts, found 1 related matches, and ran pnpm test.",
      searched: true,
      test_passed: true,
      edited: true,
    };

    expect(planRepairFollowup(output)).toEqual({
      action: "commit_repair",
      requiresHumanReview: false,
      summary: output.summary,
    });
  });

  it("asks the host to rerun tests when edits happened but verification failed", () => {
    const output: ToolRepairOutput = {
      summary: "Edited src/app.ts but the focused test failed.",
      searched: true,
      test_passed: false,
      edited: true,
    };

    expect(planRepairFollowup(output)).toEqual({
      action: "rerun_tests",
      requiresHumanReview: true,
      summary: output.summary,
    });
  });
});
